### PR TITLE
add citest script and defconfig support on sim and qemu.

### PR DESCRIPTION
**Pull request by vela-mib**
## Summary
add open posix and cmocka test cases on qemu and sim on CI 
## Impact
CI run
## Testing
local host

### CODE REVIEW

**Comment by xiaoxiang781216** [comment]
let's fix lint warning:
```
./tools/checkpatch.sh -f path/to/file
```

**Comment by no1wudi** [comment]
Please try add these line to sim-01.dat:
```
-Darwin,,sim:citest
```

**Comment by jerpelea** [comment]
please rebase and remove the merge

**Comment by vela-mib** [comment]
> please rebase and remove the merge

done

**Comment by vela-mib** [comment]
> let's fix lint warning:
> 
> ```
> ./tools/checkpatch.sh -f path/to/file
> ```

done

File: tools/ci/testrun/script/test_open_posix/test_openposix_.py
Comment (TaiJuWu) [inline tools/ci/testrun/script/test_open_posix/test_openposix_.py:1]: Could you help to split this file into small files, I think this file is too large.
You can follow the kernel directory struct or any you like.

This patch is very valuable. Thanks a lot.

Comment (xiaoxiang781216) [reply tools/ci/testrun/script/test_open_posix/test_openposix_.py:1]: let make the test run first, do in the next patch.

**Pull request merged**